txgh/lib/txgh/config/providers/gitlab_provider.rb

```diff
@@ -0,0 +1,31 @@
+module Txgh
+  module Config
+    module Providers
+      class GitlabProvider < Txgh::Config::Providers::GithubProvider
```

Comment:
Consider moving shared functionality into a base provider class like `GitProvider` and defining two subclasses for Github and Gitlab. Inheritance here is a bit weird to me.